llvm/lib/Target/AArch64/AArch64InstrInfo.cpp

```diff
@@ -8335,6 +8335,26 @@ bool AArch64InstrInfo::shouldOutlineFromFunctionByDefault(
   return MF.getFunction().hasMinSize();
 }
 
+void AArch64InstrInfo::buildClearRegister(Register Reg, MachineBasicBlock &MBB,
```

Comment:
Why make it a method on InstrInfo? Why not another method of FrameLowering?

Do you plan to call `buildClearRegister` from somewhere other than FrameLowering?

Reply:
Yes, in particular I want to call it when deciding when to clear out the register holding the stack protector guard value before exiting the function. It'll probably be during prologue/epilogue mangling, but I think the `TII` is still a nicer place to put it...

Reply:
Ah! Ok, they *why* for this PR was not clear to me.

Reply:
I added more context to the commit message.